src/Core/ConstraintValidator/AddressDniRequiredValidator.php

```diff
@@ -56,6 +57,9 @@ public function __construct(CountryRequiredFieldsProviderInterface $countryRequi
      */
     public function validate($value, Constraint $constraint)
     {
+        if (!($constraint instanceof AddressDniRequired)) {
+            return;
+        }
```

Comment:
```suggestion
    public function validate($value, AddressDniRequired $constraint)
    {
```
Why not something like this? :thinking:

Reply:
Because that brings a BC !

Reply:
I know, but not we maybe need to accept this kind of BC.

ping @matks

Reply:
IMO the "cost" to avoid the BC is very low here to pay.

As we disagree on it maybe better ask the big boss @eternoendless 😉

Reply:
Don't forget, currently if we use a different class, an exception is raised :sweat_smile:

Reply:
Oh 😮 then it's a bugfix not a BC 😄

Reply:
It's not possible : that breaks the interface `vendor/symfony/symfony/src/Symfony/Component/Validator/ConstraintValidator.php`
```php
public function validate($value, Constraint $constraint);
```

Reply:
Let's go for @Progi1984 solution. We can always improve it later. The PR should not be blocked by this minor detail.